src/towncrier/test/test_create.py

```diff
@@ -92,25 +96,56 @@ def test_content(self):
         content_line = "This is a content"
         self._test_success(content=[content_line], additional_args=["-c", content_line])
 
+    def test_content_without_eof_newline(self):
+        """
+        When creating a new fragment the content can be passed as a command line
+        argument. The text editor is not invoked, and no eof newline is added if the
+        config option is set.
+        """
+        config = dedent(
+            """\
+            [tool.towncrier]
+            package = "foo"
+            create_eof_newline = false
+            """
+        )
+        content_line = "This is a content"
+        self._test_success(
+            content=[content_line],
+            additional_args=["-c", content_line],
+            config=config,
+            eof_newline=False,
+        )
+
+    def test_content_no_eof_newline_arg(self):
+        """
+        When creating a new fragment the content can be passed as a command line
+        argument. The text editor is not invoked, and no eof newline is added if the
+        --no-eof-newline argument is passed.
+        """
+        content_line = "This is a content"
+        self._test_success(
+            content=[content_line],
+            additional_args=["-c", content_line, "--no-eof-newline"],
+            eof_newline=False,
+        )
+
     def test_message_and_edit(self):
         """
         When creating a new message, a initial content can be passed via
         the command line and continue modifying the content by invoking the
         text editor.
```

Comment:
Maybe add a comment that the conten is pass `verbatim` and no extra newline or processed is done.

Reply:
Because that comment would be incorrect. The newline option is still processed after it's received back from the editor.

Reply:
OMG :) Are people using this feature ?

I am ok with that, we can have it.

It looks so complicated :) ... but I have never felt the need for `towncrier create`

For me it's always easier to just create the file directly in my text editor, without calling towncrier.